misc-scripts/xref_mapping/XrefParser/RefSeqGPFFParser.pm

```diff
@@ -17,340 +17,330 @@ limitations under the License.
-  if (!defined $species2name{$species_id}) { return; }
-  my %species2tax  = $self->species_id2taxonomy($dbi);
-  push @{$species2tax{$species_id}}, $species_id;
-  my @names   = @{$species2name{$species_id}};
-  my @tax_ids = @{$species2tax{$species_id}};
-  my %name2species_id     = map{ $_=>$species_id } @names;
-  my %taxonomy2species_id = map{ $_=>$species_id } @tax_ids;
+# provided a params hash containing record and type, returns xref for bulk insert and creates related dependent_xrefs
+sub xref_from_record {
+  my ($self, $genbank_rec, $type) = @_;
+
+  # Get the record species
+  my ($record_species) = $genbank_rec =~ /ORGANISM\s*(.*)\n/x;
+  $record_species = lc $record_species;
+  $record_species =~ s/\s+/_/xg;
+
+  # get the record species id from the record name
+  my $record_species_id = $self->{name2species_id}->{$record_species};
+
+  # if not found try from the record taxon_id
+  if ( !defined $record_species_id ) {
+    my ($record_taxon) = $genbank_rec =~ /db_xref="taxon:(\d+)/x;
+    $record_species_id = $self->{taxonomy2species_id}->{$record_taxon};
+  }
+
+  # skip if species is not the required
+  return unless ( defined $record_species_id && ($record_species_id eq $self->{species_id}) );
 
-  # Retrieve existing RefSeq mRNA
-  my (%refseq_ids) = (%{ $self->get_valid_codes("RefSeq_mRNA", $species_id, $dbi) }, %{ $self->get_valid_codes("RefSeq_mRNA_predicted", $species_id, $dbi) });
-  my (%entrez_ids) = %{ $self->get_valid_codes("EntrezGene", $species_id, $dbi) };
-  my (%wiki_ids) = %{ $self->get_valid_codes("WikiGene", $species_id, $dbi) };
 
+  my ($acc) = $genbank_rec =~ /ACCESSION\s+(\S+)/x;
 
-  my %dependent_sources =  $self->get_xref_sources($dbi);
+  my $acc_source_id = $self->source_id_from_acc($acc);
 
-  my $add_dependent_xref_sth = $dbi->prepare("INSERT INTO dependent_xref  (master_xref_id,dependent_xref_id, linkage_source_id) VALUES (?,?, $entrez_source_id)");
+  my $xref = {
+    ACCESSION     => $acc,
+    SPECIES_ID    => $self->{species_id},
+    SOURCE_ID     => $acc_source_id,
+    SEQUENCE_TYPE => $type,
+    INFO_TYPE     => 'SEQUENCE_MATCH'
+  };
 
-  my $refseq_io = $self->get_filehandle($file);
+  my ($ver_acc, $ver_num) = $genbank_rec =~ /VERSION\s+(\w+)\.(\d+)/x;
 
-  if ( !defined $refseq_io ) {
-    print STDERR "ERROR: Can't open RefSeqGPFF file $file\n";
-    return;
+  if ($acc eq $ver_acc) {
+    $xref->{LABEL} = "${acc}.${ver_num}";
+    $xref->{VERSION} = $ver_num;
+  } else {
+    warn "WARNING: accession $acc mismatch with version ${acc}.${ver_num}\n" if $self->{verbose};
   }
 
-  my @xrefs;
+  my ($description) = $genbank_rec =~ /DEFINITION\s+  # Find the field identifier
+                                  (.*)                # get the description
+                                  \s+ACCESSION/xms;   # until the next field
 
-  local $/ = "\/\/\n";
+  # remove any newlines and spaces, and make sure is within mysql limits
+  $description =~ s/\n//xg;
+  $description =~ s/\s+/ /xg;
```

Comment:
1. This IS supposed to simply remove all whitespace, right? Because if it is supposed to replace strings of multiple whitespace characters with a single space, using /x means it does not.